### PR TITLE
catch any mismatches with the registered resolutions and resolutions in vote store at node startup

### PR DESCRIPTION
**Pull request by charithabandi**

### CODE REVIEW

**Comment by jchappelow** [comment]
Was there an issue or discussion explaining what could go wrong without this?  For instance, installing a binary that lacks the compiled-in resolution types that are referenced by the `kwild_voting.resolutions` table.  Related: #952

**Comment by brennanjl** [comment]
> Was there an issue or discussion explaining what could go wrong without this? For instance, installing a binary that lacks the compiled-in resolution types that are referenced by the kwild_voting.resolutions table. Related: https://github.com/kwilteam/kwil-db/issues/952

Yes: https://github.com/kwilteam/kwil-db/issues/842

A thread of when idOS hit this: https://kwilteam.slack.com/archives/C0578NPEU21/p1718791968399959

File: internal/voting/voting.go [hunk 1]
```diff
@@ -56,9 +56,50 @@ func initializeVoteStore(ctx context.Context, db sql.TxMaker) error {
...
+	if err != nil {
+		return fmt.Errorf("failed to validate resolution types: %w", err)
+	}
+
 	return tx.Commit(ctx)
 }
 
+// validateResolutionTypes verifies if the resolutions stored in the vote store
+// are the same as the resolutions registered. Any mismatch at the startup would
+// indicate unplanned consensus changes and should be treated as a critical error.
```
Comment (jchappelow) [inline internal/voting/voting.go:69]: `validateResolutionTypes` is called *after* a loop around `createResolutionType`, so I believe only the length check could ever hit as it is currently used.  Did we want to detect resolutions an existing table that are not in the registered resolutions map from `ListResolutions`?

File: internal/voting/voting.go [hunk 2]
```diff
@@ -41,13 +41,36 @@ func initializeVoteStore(ctx context.Context, db sql.TxMaker) error {
...
+	for _, name := range compiledResolutions {
+		resMap[name] = true
+	}
+
+	dbResolutions, err := getResolutionTypes(ctx, tx)
+	if err != nil {
+		return fmt.Errorf("failed to get resolution types: %w", err)
+	}
+
+	if len(dbResolutions) != 0 {
```
Comment (brennanjl) [inline internal/voting/voting.go:55]: Why do you check if `len(dbResolutions) != 0`?. Seems unnecessary

**Review by brennanjl — COMMENTED** [review]
Minor comment but otherwise lgtm

File: internal/voting/voting.go (hunk 2 above)
Comment (charithabandi) [reply internal/voting/voting.go]: Ah, sorry, relic from the past commits. removed it

**Review by brennanjl — APPROVED** [review]

**Pull request merged**